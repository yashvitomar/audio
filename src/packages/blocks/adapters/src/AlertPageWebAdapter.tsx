import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class AlertPageWebAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    console.log('initing AlertPageWebAdapter');
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationAlertWebMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(getName(MessageEnum.NavigationTargetMessage), 'AlertWeb');

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );

    raiseMessage.copyAllPropertiesOf(from);

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    console.log(`receive AlertPageWebAdapter==>${JSON.stringify(message)}`);
    console.log(
      `Converted AlertPageWebAdapter==>${JSON.stringify(this.convert(message))}`
    );

    this.send(this.convert(message));
  }
}
