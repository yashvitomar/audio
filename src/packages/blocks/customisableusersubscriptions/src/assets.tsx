export const imgPasswordVisible = require("../../global_assets/ic_password_visible.png");
export const imgPasswordInVisible = require("../../global_assets/ic_password_invisible.png");
export const backIcon = require("../assets/backIcon.png");
export const backArrow = require("../assets/backArrow.png")
export const crown = require("../assets/crown.png")

