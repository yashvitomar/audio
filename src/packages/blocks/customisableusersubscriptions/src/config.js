Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.validationApiContentType = "application/json";
exports.validationApiMethodType = "GET";
exports.getSubscriptionAPiEndPoint =
  "customisable_user_subscriptions/subscriptions";
exports.getSubscriptionAPiMethod = "GET";
exports.subscriptionApiContentType = "application/json";
exports.textInputPlaceHolder = "Enter Text";
exports.labelTitleText = "customisableusersubscriptions";
exports.labelBodyText = "customisableusersubscriptions Body";

exports.btnExampleTitle = "CLICK ME";
exports.subscriptions = "Subscriptions";
exports.chooseSubscription = "Choose from the list of subscriptions";
exports.subscriptionDetailsScreenName = "SubscriptionDetails";
exports.goldCoinImageUrl =
  "https://akm-img-a-in.tosshub.com/sites/btmt/images/stories/goldcoin_660_102616075839.jpg";
exports.currencySymbol = "â‚¹";
exports.zero = "0";
exports.subscribed = "Subscribed";
exports.subscribe = "Subscribe";
exports.subscriptionDetails = "Subscription Details";
exports.confirmSubscription = "Confirm Subscription";
exports.noSubscriptionAvailable = "No subscriptions available";
// Customizable Area End
