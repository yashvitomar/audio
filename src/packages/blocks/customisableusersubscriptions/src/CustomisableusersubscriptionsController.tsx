import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import { BlockComponent } from '../../../framework/src/BlockComponent';
import MessageEnum, {
  getName,
} from '../../../framework/src/Messages/MessageEnum';
import { runEngine } from '../../../framework/src/RunEngine';

// Customizable Area Start
// Customizable Area End

export const configJSON = require('./config');

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  route: any;
  data: any;
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  token: string;
  subscriptions: any;
  data: any;

  activeTab: any;
  annualPlan: boolean;
  monthlyPlan: boolean;
  currentPlan: string;
  historyData: any;
  subscriptionModal: boolean;
  // Customizable Area End
}

interface SS {
  id: any;
}

export default class CustomisableusersubscriptionsController extends BlockComponent<
  Props,
  S,
  SS
> {
  // Customizable Area Start
  getListCallId: any;
  // Customizable Area End

  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    // Customizable Area Start
    this.subScribedMessages = [
      getName(MessageEnum.NavigationPayLoadMessage),
      getName(MessageEnum.SessionResponseMessage),
      getName(MessageEnum.RestAPIRequestMessage),
      getName(MessageEnum.RestAPIResponceMessage),
    ];

    this.state = {
      token: '',
      subscriptions: [],
      data: null,
      activeTab: 'Subscription',
      annualPlan: false,
      monthlyPlan: false,
      currentPlan: 'free',
      historyData: [
        {
          planName: 'Premium Plan',
          status: 'Active',
          transactionId: '2376983',
          amount: '$57.50',
          expiryDate: 'Aug 08, 2023',
          time: '12:24 PM',
        },
        {
          planName: 'Premium Plan',
          status: 'Decline',
          transactionId: '2376983',
          amount: '$57.50',
          expiryDate: 'Aug 08, 2023',
          time: '12:24 PM',
        },
        {
          planName: 'Premium Plan',
          status: '',
          transactionId: '2376983',
          amount: '$57.50',
          expiryDate: 'Aug 08, 2023',
          time: '12:24 PM',
        },
        {
          planName: 'Premium Plan',
          status: '',
          transactionId: '2376983',
          amount: '$57.50',
          expiryDate: 'Aug 08, 2023',
          time: '12:24 PM',
        },
        {
          planName: 'Premium Plan',
          status: '',
          transactionId: '2376983',
          amount: '$57.50',
          expiryDate: 'Aug 08, 2023',
          time: '12:24 PM',
        },
      ],
      subscriptionModal: true,
    };
    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async receive(from: string, message: Message) {
    // Customizable Area Start
    runEngine.debugLog('Message Recived', message);
    if (getName(MessageEnum.NavigationPayLoadMessage) === message.id) {
      const data = message.getData(
        getName(MessageEnum.NavigationPayLoadMessage)
      );
      this.setState({ data: data });
    }
    if (getName(MessageEnum.SessionResponseMessage) === message.id) {
      let token = message.getData(getName(MessageEnum.SessionResponseToken));
      this.setState({ token: token }, () => {
        this.fetchSubscriptionsList();
      });
    }
    if (message.id === getName(MessageEnum.RestAPIResponceMessage)) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );

      const responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      if (apiRequestCallId && responseJson) {
        if (apiRequestCallId === this.getListCallId) {
          this.setState({
            subscriptions: responseJson.data,
          });
        }
      }
    }
    // Customizable Area End
  }

  async componentDidMount() {
    super.componentDidMount();
    this.getToken();
    if (this.isPlatformWeb() === false) {
      this.props.navigation.addListener('willFocus', () => {
        this.getToken();
      });
    }
    // Customizable Area Start
    // Customizable Area End
  }

  swithTab = () => {
    if (this.state.activeTab == 'Subscription') {
      this.setState({ activeTab: 'History' });
    } else {
      this.setState({
        activeTab: 'Subscription',
      });
    }
  };

  getToken = () => {
    const msg: Message = new Message(
      getName(MessageEnum.SessionRequestMessage)
    );
    this.send(msg);
  };

  fetchSubscriptionsList = () => {
    // Customizable Area Start
    const header = {
      token: this.state.token,
      'Content-Type': configJSON.subscriptionApiContentType,
    };

    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.getListCallId = requestMessage.messageId;
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.getSubscriptionAPiEndPoint
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.getSubscriptionAPiMethod
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);
    // Customizable Area End
  };

  gotoSubDetailScreen(item: any) {
    // Customizable Area Start
    const msg = new Message(getName(MessageEnum.NavigationMessage));
    msg.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'SubscriptionDetails'
    );
    msg.addData(getName(MessageEnum.NavigationPropsMessage), this.props);
    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(getName(MessageEnum.NavigationPayLoadMessage), item);
    msg.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);
    this.send(msg);
    // Customizable Area End
  }

  // Customizable Area Start
  getStatusColor = (status: string) => {
    if (status === 'Active') {
      return 'blue';
    } else if (status === 'Decline') {
      return 'red';
    } else {
      return '#000';
    }
  };
  // Customizable Area End
}
