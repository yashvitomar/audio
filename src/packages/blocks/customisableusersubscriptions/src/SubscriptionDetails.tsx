import React from "react";
  // Customizable Area Start
import {
  View,
  Text,
  ImageBackground,
  StyleSheet,
  TouchableOpacity,
  ScrollView
} from "react-native";
// Customizable Area End

import CustomisableusersubscriptionsController, {
  configJSON
} from "./CustomisableusersubscriptionsController";

export default class SubscriptionDetails extends CustomisableusersubscriptionsController {
  render() {
    return (
      // Customizable Area Start
      <View style={styles.container}>
        <ImageBackground
          style={styles.banner}
          source={{
            uri: configJSON.goldCoinImageUrl
          }}
        >
          <View style={styles.bannerContent}>
            <Text style={styles.bannerHeading}>
              {configJSON.subscriptionDetails}
            </Text>
          </View>
        </ImageBackground>
        <ScrollView
          showsVerticalScrollIndicator={false}
          style={styles.detailsContainer}
        >
          <View style={styles.planHeadingView}>
            <Text style={styles.planName} numberOfLines={1}>
              {this.state.data?.attributes?.name}
            </Text>
            <Text style={styles.price}>
              {this.state.data?.attributes?.currencySymbol ??
                configJSON.currencySymbol}
              {this.state.data?.attributes?.price ?? configJSON.zero}
            </Text>
          </View>
          <Text style={styles.dateHeading}>
              Valid Upto: {this.state.data?.attributes?.valid_up_to}
          </Text>
          <Text style={styles.detailsHeading}>
            {configJSON.subscriptionDetails}
          </Text>
          <Text style={styles.details}>
              {this.state.data?.attributes?.description}
          </Text>
          <TouchableOpacity style={styles.btnView} activeOpacity={0.7}>
            <Text style={styles.btnText}>{configJSON.confirmSubscription}</Text>
          </TouchableOpacity>
        </ScrollView>
      </View>
      // Customizable Area End
    );
  }
}

const styles = StyleSheet.create({
  // Customizable Area Start
  container: {
    flex: 1,
    backgroundColor: "#fff"
  },
  banner: {
    height: 206,
    width: "100%"
  },
  bannerContent: {
    flexDirection: "row",
    alignItems: "center",
    marginTop: 20,
    marginLeft: 15
  },
  bannerHeading: {
    fontWeight: "600",
    color: "white",
    fontSize: 22
  },
  planHeadingView: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginTop: 20
  },
  planName: {
    fontSize: 26,
    fontWeight: "700",
    maxWidth: "70%"
  },
  price: {
    fontSize: 20,
    fontWeight: "600"
  },
  detailsContainer: {
    paddingHorizontal: 15
  },
  dateHeading: {
    fontSize: 14,
    fontWeight: "700",
    marginTop: 25
  },
  detailsHeading: {
    fontSize: 14,
    fontWeight: "700",
    marginTop: 25,
    marginBottom: 20
  },
  details: {
    fontSize: 14,
    fontWeight: "500"
  },
  btnText: {
    fontWeight: "500",
    color: "#fff",
    alignSelf: "center",
    fontSize: 16
  },
  btnView: {
    height: 50,
    backgroundColor: "#cd960c",
    justifyContent: "center",
    marginTop: 50,
    marginBottom: 40
  }
  // Customizable Area End
});

// Customizable Area Start
// Customizable Area End