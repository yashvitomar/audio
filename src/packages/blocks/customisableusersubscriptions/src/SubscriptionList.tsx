import React from "react";
// Customizable Area Start
// Customizable Area End
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  FlatList,
  ImageBackground
} from "react-native";
import CustomisableusersubscriptionsController, {
  Props,
  configJSON
} from "./CustomisableusersubscriptionsController";

export class SubscriptionListView extends CustomisableusersubscriptionsController {
  constructor(props: Props) {
    super(props);
  }

  // Customizable Area Start
  SubscriptionListItem = ({
    item,
    navigate
  }: {
    item: any;
    navigate: Function;
  }) => {
    return (
      <TouchableOpacity
        style={styles.subscriptionItemContainer}
        onPress={() =>
          navigate(configJSON.subscriptionDetailsScreenName, { data: item })
        }
        activeOpacity={0.7}
      >
        <ImageBackground
          style={styles.banner}
          source={{
            uri: configJSON.goldCoinImageUrl
          }}
        >
          <View style={styles.wrapperView}>
            <View style={styles.detailsView}>
              <View>
                <Text style={styles.nameText} numberOfLines={1}>
                  {item.attributes.name}
                </Text>
                <Text style={styles.dateText}>
                  Valid upto: {item.attributes.valid_up_to}
                </Text>
              </View>
              <Text style={styles.priceText}>
                {item.attributes.currencySymbol ?? configJSON.currencySymbol}
                {item.attributes.price ?? configJSON.zero}
              </Text>
            </View>
            {item.attributes.subscribed ? (
              <TouchableOpacity
                activeOpacity={0.7}
                style={styles.subscribedBtnView}
              >
                <Text style={styles.subscribeBtnText}>
                  {configJSON.subscribed}
                </Text>
              </TouchableOpacity>
            ) : (
              <TouchableOpacity
                activeOpacity={0.7}
                style={styles.subscribeBtnView}
                onPress={() =>
                  this.gotoSubDetailScreen(item)
                }
              >
                <Text style={styles.subscribeBtnText}>
                  {configJSON.subscribe}
                </Text>
              </TouchableOpacity>
            )}
          </View>
        </ImageBackground>
      </TouchableOpacity>
    );
  };
  // Customizable Area End

  render() {
    return (
      // Customizable Area Start
      <View>
        {this.props.data && this.props.data.length > 0 ? (
          <FlatList
            style={{ marginTop: 20, paddingHorizontal: 10 }}
            data={this.props.data}
            showsVerticalScrollIndicator={false}
            //@ts-ignore
            renderItem={({ item }) => {
              return (
                <this.SubscriptionListItem
                  item={item}
                  navigate={this.props.navigation.navigate}
                />
              );
            }}
            keyExtractor={(item, index) => index.toString()}
          />
        ) : (
          <View>
            <Text>{configJSON.noSubscriptionAvailable}</Text>
          </View>
        )}
      </View>
      // Customizable Area End
    );
  }
}

const styles = StyleSheet.create({
  // Customizable Area Start
  container: {
    flex: 1,
    backgroundColor: "#f3f3f3"
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    paddingHorizontal: 10,
    paddingVertical: 20
  },
  leftHeaderView: {
    flexDirection: "row",
    alignItems: "center"
  },
  iconView: {
    paddingHorizontal: 5
  },
  backIcon: {
    width: 7,
    height: 14
  },
  myAdsText: {
    fontSize: 22,
    fontWeight: "600",
    marginLeft: 10
  },
  description: {
    fontSize: 14,
    fontWeight: "500",
    marginLeft: 15
  },

  itemContainer: {
    flex: 1
  },
  banner: {
    width: "100%",
    height: 200,
    borderRadius: 10
  },
  wrapperView: {
    width: "100%",
    height: 200,
    justifyContent: "space-between",
    padding: 15
  },
  detailsView: {
    flexDirection: "row",
    justifyContent: "space-between"
  },
  priceText: {
    fontWeight: "600",
    color: "#fff",
    fontSize: 16
  },
  statusView: {
    borderRadius: 5,
    width: 61,
    justifyContent: "center",
    alignItems: "center",
    height: 20
  },
  approvedView: {
    backgroundColor: "#50cf54",
    borderRadius: 6
  },
  rejectedView: {
    backgroundColor: "#bd3e40",
    borderRadius: 6
  },
  pendingView: {
    backgroundColor: "#e89e16"
  },
  subscriptionItemContainer: {
    width: "100%",
    marginBottom: 15
  },
  statusText: {
    fontSize: 10,
    color: "#fff"
  },
  nameText: {
    fontSize: 16,
    color: "#fff",
    fontWeight: "600",
    marginVertical: 5,
    width: 240
  },
  dateText: {
    fontSize: 12,
    color: "#fff",
    fontWeight: "600"
  },
  subscribeBtnView: {
    alignSelf: "flex-end",
    borderColor: "#fff",
    borderWidth: 1,
    paddingHorizontal: 30,
    paddingVertical: 12
  },
  subscribeBtnText: {
    color: "#fff",
    fontWeight: "600"
  },
  subscribedBtnView: {
    alignSelf: "flex-end",
    backgroundColor: "#e89e16",
    paddingHorizontal: 30,
    paddingVertical: 12
  }
  // Customizable Area End
});

// Customizable Area Start
// Customizable Area End
