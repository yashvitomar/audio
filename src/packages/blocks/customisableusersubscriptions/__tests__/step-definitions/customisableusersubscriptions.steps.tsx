import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";

import MessageEnum, {
  getName,
} from "../../../../framework/src/Messages/MessageEnum";
import React from "react";
import Customisableusersubscriptions from "../../src/Customisableusersubscriptions";
import SubscriptionDetails from "../../src/SubscriptionDetails";

const screenProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    push: jest.fn(),
    pop: jest.fn(),
    replace: jest.fn(),
  },
  id: "Customisableusersubscriptions",
  route: {},
  data: {},
};

const feature = loadFeature(
  "./__tests__/features/customisableusersubscriptions-scenario.feature"
);

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "android" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "android");
  });

  test("User navigates to customisableusersubscriptions", ({
    given,
    when,
    then,
  }) => {
    let customisableusersubscriptionsBlock: ShallowWrapper;
    let instance: Customisableusersubscriptions;

    given("I am a User loading customisableusersubscriptions", () => {
      customisableusersubscriptionsBlock = shallow(
        <Customisableusersubscriptions {...screenProps} />
      );
    });

    when("I navigate to the customisableusersubscriptions", () => {
      instance = customisableusersubscriptionsBlock.instance() as Customisableusersubscriptions;
    });

    then("customisableusersubscriptions will load with out errors", () => {
      expect(customisableusersubscriptionsBlock).toBeTruthy();
      expect(customisableusersubscriptionsBlock).toMatchSnapshot();

      const msgValidationAPI = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      msgValidationAPI.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        msgValidationAPI.messageId
      );
      msgValidationAPI.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        {
          data: [{"id":"1","type":"subscription","attributes":{"name":"Test Subscription","price":"100.0","description":"Description of the subscription","valid_up_to":"2022-02-02","expired":false,"image_link":null,"subscribed":false}},{"id":"2","type":"subscription","attributes":{"name":"iPad App","price":"500.0","description":"This is a month iPad App","valid_up_to":"2016-02-03","expired":true,"image_link":"/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBCZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--0c99d2419f59ed2dbe813a272bfb342861d45cfd/Home.png","subscribed":false}},{"id":"3","type":"subscription","attributes":{"name":"New test sub","price":"888.0","description":"Desc","valid_up_to":"2024-10-18","expired":false,"image_link":null,"subscribed":false}}]
        }
      );
      instance.getListCallId = msgValidationAPI.messageId;
      
      runEngine.sendMessage("Unit Test", msgValidationAPI);

      const msgError = new Message(
        getName(MessageEnum.RestAPIResponceErrorMessage)
      );
      msgError.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        msgValidationAPI.messageId
      );
      msgError.addData(getName(MessageEnum.RestAPIResponceErrorMessage), {
        data: []
      });
      instance.getListCallId = msgValidationAPI.messageId;
      runEngine.sendMessage("Unit Test", msgValidationAPI);

    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(customisableusersubscriptionsBlock).toBeTruthy();
      expect(customisableusersubscriptionsBlock).toMatchSnapshot();
    });
  });

  test("User navigates to subscriptiondetails", ({
    given,
    when,
    then,
  }) => {
    let subscriptionDetailsBlock: ShallowWrapper;
    let instance: SubscriptionDetails;

    given("I am a User loading subscriptiondetails", () => {
      subscriptionDetailsBlock = shallow(
        <SubscriptionDetails {...screenProps} />
      );
    });

    when("I navigate to the subscriptiondetails", () => {
      instance = subscriptionDetailsBlock.instance() as Customisableusersubscriptions;
      instance.setState({"data": [{"id":"1","type":"subscription","attributes":{"name":"Test Subscription","price":"100.0","description":"Description of the subscription","valid_up_to":"2022-02-02","expired":false,"image_link":null,"subscribed":false}},{"id":"2","type":"subscription","attributes":{"name":"iPad App","price":"500.0","description":"This is a month iPad App","valid_up_to":"2016-02-03","expired":true,"image_link":"/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBCZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--0c99d2419f59ed2dbe813a272bfb342861d45cfd/Home.png","subscribed":false}},{"id":"3","type":"subscription","attributes":{"name":"New test sub","price":"888.0","description":"Desc","valid_up_to":"2024-10-18","expired":false,"image_link":null,"subscribed":false}}]});
    });

    then("subscriptiondetails will load with out errors", () => {
      expect(subscriptionDetailsBlock).toBeTruthy();
      expect(subscriptionDetailsBlock).toMatchSnapshot();

      const payloadMessage = new Message(
        getName(MessageEnum.NavigationPayLoadMessage)
      );
      runEngine.sendMessage("Unit Test", payloadMessage);

    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(subscriptionDetailsBlock).toBeTruthy();
      expect(subscriptionDetailsBlock).toMatchSnapshot();
    });
  });

});
