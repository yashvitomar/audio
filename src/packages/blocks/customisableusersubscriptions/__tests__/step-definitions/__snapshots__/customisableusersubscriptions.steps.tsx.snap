// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customisableusersubscriptions User navigates to customisableusersubscriptions 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.7}
          onPress={[Function]}
          style={
            Object {
              "paddingHorizontal": 5,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../packages/blocks/customisableusersubscriptions/assets/backIcon.png",
              }
            }
            style={
              Object {
                "height": 14,
                "width": 7,
              }
            }
          />
        </TouchableOpacity>
        <Text
          style={
            Object {
              "fontSize": 22,
              "fontWeight": "600",
              "marginLeft": 10,
            }
          }
        >
          Subscriptions
        </Text>
        <Text />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "marginLeft": 15,
        }
      }
    >
      Choose from the list of subscriptions
    </Text>
    <SubscriptionListView
      data={Array []}
      id="Customisableusersubscriptions"
      navigation={
        Object {
          "goBack": [MockFunction],
          "navigate": [MockFunction],
          "pop": [MockFunction],
          "push": [MockFunction],
          "replace": [MockFunction],
        }
      }
      route={Object {}}
    />
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`customisableusersubscriptions User navigates to customisableusersubscriptions 2`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.7}
          onPress={[Function]}
          style={
            Object {
              "paddingHorizontal": 5,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../packages/blocks/customisableusersubscriptions/assets/backIcon.png",
              }
            }
            style={
              Object {
                "height": 14,
                "width": 7,
              }
            }
          />
        </TouchableOpacity>
        <Text
          style={
            Object {
              "fontSize": 22,
              "fontWeight": "600",
              "marginLeft": 10,
            }
          }
        >
          Subscriptions
        </Text>
        <Text />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "marginLeft": 15,
        }
      }
    >
      Choose from the list of subscriptions
    </Text>
    <SubscriptionListView
      data={
        Array [
          Object {
            "attributes": Object {
              "description": "Description of the subscription",
              "expired": false,
              "image_link": null,
              "name": "Test Subscription",
              "price": "100.0",
              "subscribed": false,
              "valid_up_to": "2022-02-02",
            },
            "id": "1",
            "type": "subscription",
          },
          Object {
            "attributes": Object {
              "description": "This is a month iPad App",
              "expired": true,
              "image_link": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBCZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--0c99d2419f59ed2dbe813a272bfb342861d45cfd/Home.png",
              "name": "iPad App",
              "price": "500.0",
              "subscribed": false,
              "valid_up_to": "2016-02-03",
            },
            "id": "2",
            "type": "subscription",
          },
          Object {
            "attributes": Object {
              "description": "Desc",
              "expired": false,
              "image_link": null,
              "name": "New test sub",
              "price": "888.0",
              "subscribed": false,
              "valid_up_to": "2024-10-18",
            },
            "id": "3",
            "type": "subscription",
          },
        ]
      }
      id="Customisableusersubscriptions"
      navigation={
        Object {
          "goBack": [MockFunction],
          "navigate": [MockFunction],
          "pop": [MockFunction],
          "push": [MockFunction],
          "replace": [MockFunction],
        }
      }
      route={Object {}}
    />
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`customisableusersubscriptions User navigates to subscriptiondetails 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "uri": "https://akm-img-a-in.tosshub.com/sites/btmt/images/stories/goldcoin_660_102616075839.jpg",
      }
    }
    style={
      Object {
        "height": 206,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 22,
            "fontWeight": "600",
          }
        }
      >
        Subscription Details
      </Text>
    </View>
  </ImageBackground>
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "paddingHorizontal": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 20,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "fontSize": 26,
            "fontWeight": "700",
            "maxWidth": "70%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "600",
          }
        }
      >
        â‚¹
        0
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "700",
          "marginTop": 25,
        }
      }
    >
      Valid Upto: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "700",
          "marginBottom": 20,
          "marginTop": 25,
        }
      }
    >
      Subscription Details
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.7}
      style={
        Object {
          "backgroundColor": "#cd960c",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 40,
          "marginTop": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Confirm Subscription
      </Text>
    </TouchableOpacity>
  </ScrollViewMock>
</View>
`;

exports[`customisableusersubscriptions User navigates to subscriptiondetails 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "uri": "https://akm-img-a-in.tosshub.com/sites/btmt/images/stories/goldcoin_660_102616075839.jpg",
      }
    }
    style={
      Object {
        "height": 206,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 22,
            "fontWeight": "600",
          }
        }
      >
        Subscription Details
      </Text>
    </View>
  </ImageBackground>
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "paddingHorizontal": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 20,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "fontSize": 26,
            "fontWeight": "700",
            "maxWidth": "70%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "600",
          }
        }
      >
        â‚¹
        0
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "700",
          "marginTop": 25,
        }
      }
    >
      Valid Upto: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "700",
          "marginBottom": 20,
          "marginTop": 25,
        }
      }
    >
      Subscription Details
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.7}
      style={
        Object {
          "backgroundColor": "#cd960c",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 40,
          "marginTop": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Confirm Subscription
      </Text>
    </TouchableOpacity>
  </ScrollViewMock>
</View>
`;
